### DISCUSSION TITLE
IDE to use

### DISCUSSION THREAD

**Discussion by night2876** [xoreos/xoreos#587] [Q&A]
Hi,
i've downloaded your project sourcecode with git and integrated all the system with the fantastic wiki. All go ok in the MSYS2 and Mingw64 environment. Autotools and make build the code without problem ... very slow. Ok now if i want to edit/debug code to learn how work which IDE you use. I tryed with Visual Studio Code, add C/C++ extension but can't add the header file and the library used by the project, example ZLIB etc. Same problem with the cmake extension and the script used to assemble executable externaly from the MSYS2 environment.

Any suggestion? I must use an integrated environment like Linux or MacOS to try to develop something?

Best regards.

**Comment by DrMcCoy** [comment]
Frankly, I for one don't use an IDE as such. I use [vim](https://en.wikipedia.org/wiki/Vim_(text_editor)), on Linux, in [urxvt](https://en.wikipedia.org/wiki/Rxvt). For debugging, I mainly use [gdb](https://en.wikipedia.org/wiki/Gdb).

While I'm personally not a fan of it, you should be able to set up something like [JetBrains CLion](https://en.wikipedia.org/wiki/CLion) with xoreos no problem, by using the CMake file. IIRC, at least one person who contributed to xoreos in the past used that.

I'm not sure how it works on Windows though. I generally find Windows very hostile to developers, to be honest. Especially when it comes to library dependencies. YMMV, though.

As for Visual Studio, it generally doesn't work together with MSYS2. MSYS2 is for when you want to use a Unix-like environment, together with mingw/gcc (or clang), while Visual Studio wants to use its own environment and the Microsoft compiler. Being a Linux person, MSYS2 is generally what I want and use when I'm forced to use Windows and produce Windows binaries on Windows.

If you really want to use Visual Studio, you'd use the CMake file to generate a project file which you'd load in Visual Studio, yes. But you have to take care of the library dependencies yourself, then. You can't use the ones from MSYS2, because those have been compiled with gcc, against the mingw libc and STL.

The traditional Windows way is to collect them manually yourself, which is pretty annoying and horrible (which is why I'm not a fan of Windows as a dev environment), but in recent years, there has been several hacky solutions for that, one of them being [vcpkg](https://en.wikipedia.org/wiki/Vcpkg). I'm using that for the [MSVC workflow in GitHub actions](https://github.com/xoreos/xoreos/blob/master/.github/workflows/windows_cmake_msvc.yml) ([the vcpkg.json listing the dependencies is here](https://github.com/xoreos/xoreos/blob/master/.github/workflows/vcpkg.json)), for example. As you can see in the workflow file, I still need to manually take care of xvid, because that's not available in vcpkg, but at least they offer precompiled binaries for MSVC.

When you have figured out how to set up Visual Studio for xoreos so that it makes sense to people who prefer to develop on Windows with Visual Studio, feel free to send detailed instructions our way, so we can put them up on our Wiki for other people. I'd appreciate it, thanks!

Likewise, we'll also take instructions for other IDEs, like CLion, if you go that way.

**Reply by Lubieerror** [reply]
IMHO since Visual Studio doesn't like gcc/clang and stuff, I think the better choice might be Visual Studio Code ([Using GCC with MinGW](https://code.visualstudio.com/docs/cpp/config-mingw)).
I might try to check both JBCL and VSC in my free time.

**Reply by night2876** [reply]
I tryed that but the include file and the library, don't work. I include the path tu all the include configuring the c/c++ extension and set the lib path path cmake say no zlib found and stop to execute ... so now i'm trying with a different approch using WSL linux distribution in windows 10 and Visual Studio Code or CLion.

Best regards.
